File: solutions/easy/3019_Number_of_Changing_Keys.py

from utils.test_driver import test_driver

class Solution:
    def countKeyChanges(self, s: str) -> int:
        last_key_lower='#'
        res=-1
        for c in s:
            c=str.lower(c)
            if c!=last_key_lower:res+=1
            last_key_lower=c
        return res
if __name__ == "__main__":
    s = Solution()

    tests = [
        [
            # inputs
            [
                "aAbBcC"
            ],
            # res
            2

        ],
        [
            # inputs
            [
                "aAAAAAAAAAaaaaaa"
            ],
            # res
            0

        ],
        




    ]
    for input, res in tests:
        test_driver(s.countKeyChanges, input[0],  expected=res)
# greedy

File: solutions/easy/3019_Number_of_Changing_Keys.cpp
#include "utils/linked_list.hpp"
#include "utils/binary_tree.hpp"
#include <iostream>
#include <queue>
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <cstring>
#include <math.h>
#include <map>
using namespace binary_tree;
using namespace std;
/**copy from here**/
class Solution {
public:
    int countKeyChanges(string s) {
        int last_key=-1,res=0;
		for (auto c :s){
			int cur=c;
			if (last_key!=cur){
				if (abs(last_key-cur)!=32)res++;
			}
			
			last_key=cur;
		}
		return res-1;
    }
};
/**copy to here**/
int main()
{
	Solution *s = new Solution();
	string a;
	int res;

	a="aAbBcC";
	res=s->countKeyChanges(a);
	cout<<res<<endl;//3

	

	
	return 0;
}

